### PR TITLE
boards: arm: add ADC support for STM32 boards: nucleo_f429zi stm32f3_disco nucleo_g474re disco_l475_iot1 nucleo_l4r5zi

### PR DESCRIPTION
**Pull request by ABOSTM**
Add ADC support for : 
* nucleo_f429zi 
* stm32f3_disco
* nucleo_g474re
* disco_l475_iot1
* nucleo_l4r5zi

Tested with tests/drivers/adc/adc_api/

### CODE REVIEW

File: boards/arm/nucleo_f429zi/nucleo_f429zi.dts [hunk 1]
```diff
@@ -53,6 +53,11 @@
 	};
 };
 
+&adc1 {
+	status = "okay";
+};
+
+
```
Comment (erwango) [inline boards/arm/nucleo_f429zi/nucleo_f429zi.dts:60]: Remove one of the 2 blank lines here

**Review by erwango — CHANGES_REQUESTED** [review]
One minor comment, otherwise LGTM

File: boards/arm/nucleo_f429zi/nucleo_f429zi.dts (hunk 1 above)
Comment (ABOSTM) [reply boards/arm/nucleo_f429zi/nucleo_f429zi.dts]: Done

**Review by erwango — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Comment by ABOSTM** [comment]
Restart CI

**Comment by ABOSTM** [comment]
Rebased

**Comment by ABOSTM** [comment]
Buildkite issue is independent from this PR.
This PR just highlight an existing issue: #27753

**Comment by ABOSTM** [comment]
@carlescufi , @nashif , @ydamigos 
Could this PR be merged (for next release) despite build kite issue? knowing that buildkite issue is independent from this PR and I opened dedicated issue for that #27753

**Comment by ABOSTM** [comment]
Rebased.
@carlescufi , @nashif , @ydamigos
Could this PR be merged (for next release) despite build kite issue? knowing that buildkite issue is independent from this PR and I opened dedicated issue for that #27753

**Comment by carlescufi** [comment]
> Rebased.
> @carlescufi , @nashif , @ydamigos
> Could this PR be merged (for next release) despite build kite issue? knowing that buildkite issue is independent from this PR and I opened dedicated issue for that #27753

If I do I might be breaking master for everyone. I have rerun CI now and I will evaluate the results once it's done.

**Comment by ABOSTM** [comment]
> If I do I might be breaking master for everyone. I have rerun CI now and I will evaluate the results once it's done.

I understand, but note that issue #27753 highlighted by buildkite is already present in master even if this PR is not merged.

**Comment by ABOSTM** [comment]
Rebase and restart buildkite

**Comment by ABOSTM** [comment]
Now that #27753 is fixed (lis2dh sensor), buildkite highlight similar issue on hts221 and lsm6dsl sensors.
I submit new issue for that: #28390

**Review by FRASTM — APPROVED** [review]

**Comment by ABOSTM** [comment]
Restart buildkite

**Comment by ABOSTM** [comment]
Now Buildkite is successful, PR can be merged.  
... for 2.4.0 ?
@carlescufi @MaureenHelm

**Comment by MaureenHelm** [comment]
> Now Buildkite is successful, PR can be merged.
> ... for 2.4.0 ?
> @carlescufi @MaureenHelm

It's not a bug fix or documentation, so it needs to wait until after the 2.4 release.

**Pull request merged**